### PR TITLE
add interface for ignoring identity through command line

### PR DESCRIPTION
**Pull request by nirvanagit**
## Changes:
Add interface for endpoint generation suspension so that other implementations can be easily swapped with the current command line implementation.

### CODE REVIEW

File: admiral/pkg/clusters/endpointSuspender.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package clusters
+
+type EndpointSuspender interface {
```
Comment (aattuluri) [inline admiral/pkg/clusters/endpointSuspender.go:3]: Refactor this to EndpointUpdateSuspender?

File: admiral/pkg/clusters/endpointSuspension.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+func (des *endpointSuspension) identityByEnvironment(identity, environment string) bool {
+	log.Printf("op=dynamicEndpointSuspension message=checking if identity %s in environment %s is in the suspension list",
+		identity, environment)
+	des.ignoredIdentityCache.RWLock.RLock()
+	defer des.ignoredIdentityCache.RWLock.RUnlock()
+	if des.ignoredIdentityCache.EnvironmentsByIdentity[identity] != nil {
+		identityEnvironments := des.ignoredIdentityCache.EnvironmentsByIdentity[identity]
+		if len(identityEnvironments) == 0 || (len(identityEnvironments) == 1 && identityEnvironments[0] == "") {
+			log.Printf("%s, identity: %s", alertMsgSuspensionForIdentityInAllEnvironments, identity)
```
Comment (aattuluri) [inline admiral/pkg/clusters/endpointSuspension.go:79]: `(len(identityEnvironments) == 1 && identityEnvironments[0] == "")` is this a special case or just a safety check?

**Review by aattuluri — CHANGES_REQUESTED** [review]
lgtm, a small change requested and a question.

File: admiral/pkg/clusters/endpointSuspension.go (hunk 2 above)
Comment (nirvanagit) [reply admiral/pkg/clusters/endpointSuspension.go:79]: Yes, this is the case when the entry is like:
```golang
identityByEnvironment: map[string][]string{
  "identity": []string{""},
}
```

File: admiral/pkg/clusters/endpointSuspender.go (hunk 1 above)
Comment (nirvanagit) [reply admiral/pkg/clusters/endpointSuspender.go:3]: How about `ServiceEntryUpdateSuspender` ?

**Review by aattuluri — APPROVED** [review]
lgtm

**Pull request merged**